extensions/resource-deployment/src/main.ts

```diff
@@ -27,7 +29,7 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 		const errorMessage = localize('resourceDeployment.FailedToLoadExtension', "Failed to load extension: {0}, Error detected in the resource type definition in package.json, check debug console for details.", context.extensionPath);
 		vscode.window.showErrorMessage(errorMessage);
 		validationFailures.forEach(message => console.error(message));
-		return;
+		return <rd.IExtension><unknown>undefined;
```

Comment:
You can just cast to any, we don't care about the type when we're just returning it. `return <any>undefined`. Although realistically I don't like that we return here - I would prefer that we do one of two options : 

1. Throw an error so the extension fails to activate
2. Still load any valid resource types and finish activating the extension, but display a warning about the ones that failed to load.

Anyways for now returning undefined is ok so we keep the same-ish behavior, but please make a work item to follow up on this.

Reply:
change to <any> cast. Yes, we should throw here, but I did not want to introduce that behavior change in this PR.